CMakeLists.txt

```diff
@@ -112,6 +95,42 @@ configure_package_config_file(
     INSTALL_DESTINATION "${CONFIG_INSTALL_DIR}"
 )
 
+# zip
+set(Header_Files "src/miniz.h" "src/zip.h")
+source_group("Header Files" FILES "${Header_Files}")
+
+set(Source_Files "src/zip.c")
+source_group("Source Files" FILES "${Source_Files}")
+
+add_library(${PROJECT_NAME} "${Header_Files}" "${Source_Files}")
+target_link_libraries("${PROJECT_NAME}" PUBLIC "${PROJECT_NAME}_compiler_flags")
+set_target_properties("${PROJECT_NAME}" PROPERTIES LINKER_LANGUAGE C)
+
+add_library(${NAMESPACE}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+
+if(ZIP_STATIC_PIC)
+    set_property(TARGET ${PROJECT_NAME} PROPERTY POSITION_INDEPENDENT_CODE 1)
+endif()
+
+set_property(TARGET ${PROJECT_NAME} PROPERTY C_VISIBILITY_PRESET hidden)
+if(BUILD_SHARED_LIBS)
+    target_compile_definitions(${PROJECT_NAME}
+            PUBLIC ZIP_SHARED
+            PRIVATE ZIP_BUILD_SHARED
+    )
+endif()
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+        "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/src>"
+        "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+        "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}>"
```

Comment:
.. but now there are two. This will create variations in usage and eventually make it harder to track user errors.